src/rest-server/src/models/v2/job/k8s.js

```diff
@@ -973,22 +875,55 @@ const get = async (frameworkName) => {
       },
     ],
   });
-  if (framework) {
-    const frameworkDetail = await convertFrameworkDetail(
-      JSON.parse(framework.snapshot),
-      framework.tags,
-    );
-    frameworkDetail.jobStatus.submissionTime = new Date(
-      framework.submissionTime,
-    ).getTime();
-    return frameworkDetail;
-  } else {
+
+  if (!framework) {
     throw createError(
       'Not Found',
       'NoJobError',
       `Job ${frameworkName} is not found.`,
     );
   }
+
+  const snapshot = JSON.parse(framework.snapshot);
+
+  // find corresponding job attempt when specified
+  if (jobAttemptId !== undefined) {
+    if (jobAttemptId < snapshot.status.attemptStatus.id) {
+      const frameworkHistory = await databaseModel.FrameworkHistory.findOne({
+        attributes: ['snapshot'],
+        where: {
+          frameworkName: encodeName(frameworkName),
+          attemptIndex: jobAttemptId,
+        },
+      });
+      if (!frameworkHistory) {
+        throw createError(
+          'Not Found',
+          'NoJobError',
+          `JobAttemptId ${jobAttemptId} is not found in ${frameworkName}.`,
+        );
+      }
+      // replace `attemptStatus`
+      snapshot.status.attemptStatus = JSON.parse(
+        frameworkHistory.snapshot,
+      ).status.attemptStatus;
```

Comment:
Only replace attemptStatus is not enough, you need to do merge like convert task.

Reply:
> Only replace attemptStatus is not enough, you need to do merge like convert task.

Fixed, thanks